compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -305,12 +305,7 @@ impl<'tcx> Const<'tcx> {
             // mir.
             match tcx.at(expr.span).lit_to_const(lit_input) {
                 Ok(c) => return Some(c),
-                Err(e) => {
-                    tcx.dcx().span_delayed_bug(
-                        expr.span,
-                        format!("Const::from_anon_const: couldn't lit_to_const {e:?}"),
-                    );
-                }
+                Err(_) => return None,
```

Comment:
Could you make it so we only return `None` if the expected type from `lit_input` contains aliases? That should be checkable via the `.has_aliases()` method. Then we can keep ICEing in other cases while still fixing the bug which should be "safer".

Reply:
Done :)